File: 0946-validate-stack-sequences/0946-validate-stack-sequences.py
class Solution:
    def validateStackSequences(self, pushed: List[int], popped: List[int]) -> bool:
        st=[]
        j=0
        for i in pushed:
            st.append(i)
            while st and st[-1]==popped[j]:
                st.pop()
                j+=1
        if st:return False
        return True
        
                
        

File: 0946-validate-stack-sequences/0946-validate-stack-sequences.cpp
class Solution {
public:
    bool validateStackSequences(vector<int>& pushed, vector<int>& popped) {
        stack<int>st;
        int j=0;
        for(auto c: pushed){
            st.push(c);
            while(st.size()>0 && st.top()==popped[j]){
                st.pop();
                j++;
            }
        }
        return st.size()==0;
    }
};